modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -1339,8 +1344,10 @@ NSFVAction::addINSMomentumFrictionKernels()
             params.set<MooseFunctorName>("quadratic_coef_name") = _friction_coeffs[block_i][type_i];
         }
 
-        _problem->addFVKernel(
-            kernel_type, "ins_momentum_friction_" + block_name + "_" + NS::directions[d], params);
+        _problem->addFVKernel(kernel_type,
+                              "ins_momentum_friction_" + std::to_string(block_i) + "_" +
```

Comment:
I liked block_name a lot, I would rather we do not use an arbitrary index that wont correspond to the block_id even

Reply:
Restored block_name!